metadata-ingestion/src/datahub/ingestion/source/hex/datahub.py

```diff
@@ -0,0 +1,385 @@
+                    QueryPropertiesClass.ASPECT_NAME,
+                    QuerySubjectsClass.ASPECT_NAME,
+                ],
+                with_system_metadata=False,
+            )
+            self.report.num_calls_fetch_query_entities += 1
+            logger.debug(f"Get entities v2 response: {response_json}")
+            batch_entities_by_urn = EntitiesResponse.parse_obj(response_json)
+            entities_by_urn.update(batch_entities_by_urn)
+
+        return entities_by_urn
+
+    def _fetch_query_urns_filter_hex_and_last_modified(self) -> List[QueryUrn]:
+        """
+        This could be implemented as follows, but search sdk still requires some fixes.
+        query_urns = self.datahub_client.search.get_urns(
+            filter=F.and_(
+                F.entity_type(QueryUrn.ENTITY_TYPE),
+                F.custom_filter("origin", "EQUAL", [HEX_PLATFORM_URN.urn()]),
+                F.custom_filter(
+                    "lastModifiedAt",
+                    "GREATER_THAN",
+                    [str(last_modified_at_millis)],
+                ),
+            ),
+        )
+        """
+        last_modified_at_millis = int(self.start_datetime.timestamp() * 1000)
+
+        query_urn_strs = self.datahub_client._graph.get_urns_by_filter(
+            entity_types=[QueryUrn.ENTITY_TYPE],
+            batch_size=self.page_size,
+            extraFilters=[
+                {
+                    "field": "origin",
+                    "condition": "EQUAL",
+                    "values": [HEX_PLATFORM_URN.urn()],
+                    "negated": False,
+                },
+                {
+                    "field": "lastModifiedAt",
+                    "condition": "GREATER_THAN",
+                    "values": [
+                        last_modified_at_millis,
+                    ],
+                    "negated": False,
+                },
+            ],
+        )
+        query_urns = [QueryUrn.from_string(urn_str) for urn_str in query_urn_strs]
+        logger.debug(f"Get URNS by filter: {query_urns}")
+        return query_urns
+
+    def _extract_hex_metadata(self, sql_statement: str) -> Optional[tuple[str, str]]:
+        """
+        Extract project ID and workspace name from SQL statement.
+
+        Looks for Hex metadata in SQL comments in the format:
+        -- Hex query metadata: {"project_id": "...", "project_url": "https://app.hex.tech/{workspace_name}/hex/..."}
+
+        Example:
+        -- Hex query metadata: {"categories": ["Scratchpad"], "cell_type": "SQL", "connection": "Long Tail Companions", "context": "SCHEDULED_RUN", "project_id": "d73da67d-c87b-4dd8-9e7f-b79cb7f822cf", "project_url": "https://app.hex.tech/acryl-partnership/hex/d73da67d-c87b-4dd8-9e7f-b79cb7f822cf/draft/logic?selectedCellId=67c38da0-e631-4005-9750-5bdae2a2ef3f"}
+
+        # TODO: Consider supporting multiline metadata format in the future:
+        # -- Hex query metadata: {
+        # --   "categories": ["Scratchpad"],
```

Comment:
I think Hex guarantees that the query metadata will be on a single line. It also should be true that the Hex metadata is the only thing on that line.

Given that, I think this parsing code could be a lot simpler

Reply:
Supporting multiline metadata is just a comment to explicitly mention that this is not supported right now. So there is no additional complexity in current implementation because of that.

```
TODO: Consider supporting multiline metadata format in the future
```

I agree implementation of this method could be much more simple with a simple regex.